main.go

```diff
@@ -15,6 +15,138 @@
 
 package main
 
+import (
+	"errors"
+	"fmt"
+	"log/slog"
+	"os"
+	"os/exec"
+	"os/signal"
+	"syscall"
+	"time"
+
+	"github.com/spf13/cast"
+
+	"github.com/bank-vaults/secret-init/logger"
+	"github.com/bank-vaults/secret-init/providers"
+	"github.com/bank-vaults/secret-init/providers/vault"
+)
+
+var providersMap = map[string]providers.Provider{
+	// "aws": aws.NewAWSProvider(),
+	// "gcp": gcp.NewGCPProvider(),
+	"vault": vault.NewVaultProvider(),
+}
+
 func main() {
-	println("hello world")
+	logger := logger.SetupSlog()
+
+	var providerName string
+	if len(os.Args) >= 3 && os.Args[1] == "-p" {
+		providerName = os.Args[2]
+
+		// Remove the "-p {provider name}" argument from the command-line arguments
+		os.Args = append(os.Args[:1], os.Args[3:]...)
+	}
+
+	provider, found := providersMap[providerName]
+	if !found {
+		logger.Error("invalid provider specified.", slog.String("provider name", providerName))
+
+		os.Exit(1)
+	}
+
+	if len(os.Args) == 1 {
+		logger.Error("no command is given, vault-env can't determine the entrypoint (command), please specify it explicitly or let the webhook query it (see documentation)")
+
+		os.Exit(1)
+	}
+
+	daemonMode := cast.ToBool(os.Getenv("VAULT_ENV_DAEMON"))
+	delayExec := cast.ToDuration(os.Getenv("VAULT_ENV_DELAY"))
```

Comment:
note: it would be useful to indicate why this is loaded from env instead of cli

Reply:
vault-env historically did not take any arguments to make sure all arguments passed to vault-env are part of the downstream command.

The above flag already changes that behavior in a backward incompatible way. We could use the -- pattern to signal where the actual downstream command arg list starts.

But it may be better to stick to env vars for now.

Note: take a note of the env vars. They are not vault specific, yet are prefixed.